streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -118,7 +118,9 @@ public class StreamTask extends AbstractTask implements ProcessorNodePunctuator,
     private boolean commitNeeded = false;
     private boolean commitRequested = false;
     private boolean hasPendingTxCommit = false;
-    private Optional<Long> timeCurrentIdlingStarted;
+    private Optional<Long> timeCurrentIdlingStarted; // time since the task started idling, empty if not idling
+    private long lastNotReadyLogTime = -1L;  // -1 means ready (no logging needed), >= 0 means not ready and tracks last log time
```

Comment:
Nit: we use `Optional<Long>` for `timeCurrentIdlingStarted` -- should we do the same for `lastNotReadyLogTime` to avoid using two different pattern?